File: Dijkstra.jl
const INF = typemax(Int)

function dijkstra(graph::Array{Int,2}, start::Int)
    n = size(graph, 1)
    dist = fill(INF, n)
    visited = fill(false, n)

    dist[start] = 0

    for _ in 1:n-1
        minDist = INF
        u = -1

        # Trouver le sommet non visité avec la distance minimale
        for i in 1:n
            if !visited[i] && dist[i] < minDist
                minDist = dist[i]
                u = i
            end
        end

        if u == -1
            break
        end

        visited[u] = true

        # Mise à jour des distances
        for v in 1:n
            if !visited[v] && graph[u, v] != 0 &&
               dist[u] + graph[u, v] < dist[v]
                dist[v] = dist[u] + graph[u, v]
            end
        end
    end

    return dist
end

# Exemple d'utilisation
graph = [
    0 4 0 0 6;
    4 0 2 0 0;
    0 2 0 5 0;
    0 0 5 0 3;
    6 0 0 3 0
]

start = 1
distances = dijkstra(graph, start)

println("Distances depuis le sommet ", start)
for i in 1:length(distances)
    println("Vers ", i, " : ", distances[i])
end

File: Dijkstra.lua
local INF = math.huge

function dijkstra(graph, start)
    local n = #graph
    local dist = {}
    local visited = {}

    -- Initialisation
    for i = 1, n do
        dist[i] = INF
        visited[i] = false
    end

    dist[start] = 0

    for _ = 1, n - 1 do
        local minDist = INF
        local u = -1

        -- Trouver le sommet non visité avec la distance minimale
        for i = 1, n do
            if not visited[i] and dist[i] < minDist then
                minDist = dist[i]
                u = i
            end
        end

        if u == -1 then break end

        visited[u] = true

        -- Mise à jour des distances
        for v = 1, n do
            if not visited[v] and graph[u][v] ~= 0
               and dist[u] + graph[u][v] < dist[v] then
                dist[v] = dist[u] + graph[u][v]
            end
        end
    end

    return dist
end

-- Exemple d'utilisation
local graph = {
    {0, 3, 0, 0, 5},
    {3, 0, 2, 0, 0},
    {0, 2, 0, 4, 0},
    {0, 0, 4, 0, 1},
    {5, 0, 0, 1, 0}
}

local start = 1
local distances = dijkstra(graph, start)

print("Distances depuis le sommet " .. start)
for i = 1, #distances do
    print("Vers " .. i .. " : " .. distances[i])
end
